packages/calcite-components/src/tests/commonTests.ts

```diff
@@ -1376,3 +1380,165 @@ export async function t9n(componentTestSetup: ComponentTestSetup): Promise<void>
     await page.waitForChanges();
   }
 }
+
+/**
+ * Helper to test openClose component setup.
+ *
+ * Note that this helper should be used within a `describe` block.
+ *
+ * @example
+ * import { openClose } from "../../tests/commonTests";
+ *
+ * openClose("calcite-tooltip", "open", false, {
+ *   open: async (page) => {
+ *     await page.keyboard.press("Tab");
+ *     await page.waitForChanges();
+ *   },
+ *   close: async (page) => {
+ *     await page.keyboard.press("Tab");
+ *     await page.waitForChanges();
+ *   },
+ * });
+ *
+ * @param {string} componentTagOrHTML - the component tag or HTML markup to test against
+ * @param {string} toggleProp - Toggle property to test. Currently, either "open" or "expanded".
+ * @param {boolean} initialToggleValue - Indicates the initial value of the toggle property to determine whether to configure a `simplePageSetup` or `newProgrammaticE2EPage`.
+ * @param {userInputDevice} userInputDevice - Optional argument with functions to simulate user input (mouse or keyboard), to open or close the component.
+ */
+
+type userInputDevice = {
+  open: (page: E2EPage) => Promise<void>;
+  close: (page: E2EPage) => Promise<void>;
+};
+
+export async function openClose(
+  componentTagOrHTML: TagOrHTML,
+  toggleProp = "open",
```

Comment:
`open` is the default, but it is still being specified in all tests. 

Most of our test helpers provide arguments for required values, and extra options are provided in an options parameters. I think we should do this to simplify this helper's usage:

```ts
interface OpenCloseOptions {
  /**
   * ...
   */
  initialToggleValue?: boolean;
  
  /**
   * ...
   */
  toggleProp?: string;

  /**
   * ...
   */
  userInputDevice?: UserInputDevice;
}

export async function openClose(
  componentTagOrHTML: TagOrHTML,
  options?: OpenCloseOptions
)
```

The main benefit we get from this is that we avoid passing defaults when we need to use `userInputDevice`:

```ts
openClose("calcite-magic", {
  userInputDevice: { /* ... */ }
});
```

vs

```ts
openClose("calcite-magic", "open", false, {
  userInputDevice: { /* ... */ }
});
```

Reply:
On a related note, I think some parameter names could be renamed to better convey their intent. This is just me, but this is how I'm currently reading them:

* `toggleProp` – most likely from `toggle`, but this can read as a boolean to enable toggling
* `initialToggleValue` – the initial value before toggling takes place or to toggle with
* `userInputDevice` – an object to specify how the user will interact with the component

Let's sync up on this to hash out some names.